x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/services/analytics_service/delete_analytics.ts

```diff
@@ -39,3 +36,88 @@ export const deleteAnalytics = async (d: DataFrameAnalyticsListRow) => {
   }
   refreshAnalyticsList$.next(REFRESH_ANALYTICS_LIST_STATE.REFRESH);
 };
+
+export const deleteAnalyticsAndTargetIndex = async (
+  d: DataFrameAnalyticsListRow,
+  deleteTargetIndex: boolean,
+  deleteTargetPattern: boolean
+) => {
+  const toastNotifications = getToastNotifications();
+  const destinationIndex = d.config.dest.index;
+  try {
+    if (isDataFrameAnalyticsFailed(d.stats.state)) {
+      await ml.dataFrameAnalytics.stopDataFrameAnalytics(d.config.id, true);
+    }
+    const status = await ml.dataFrameAnalytics.deleteDataFrameAnalyticsAndTargetIndex(
+      d.config.id,
+      deleteTargetIndex,
+      deleteTargetPattern
+    );
+    if (status.acknowledged) {
+      toastNotifications.addSuccess(
+        i18n.translate('xpack.ml.dataframe.analyticsList.deleteAnalyticsSuccessMessage', {
+          defaultMessage: 'Request to delete data frame analytics {analyticsId} acknowledged.',
+          values: { analyticsId: d.config.id },
+        })
+      );
+    }
+
+    if (status.deleteTargetIndexAcknowledged) {
+      toastNotifications.addSuccess(
+        i18n.translate('xpack.ml.dataframe.analyticsList.deleteAnalyticsWithIndexSuccessMessage', {
+          defaultMessage: 'Request to delete destination index {destinationIndex} acknowledged.',
+          values: { destinationIndex },
+        })
+      );
+    }
+
+    if (status.deleteIndexPatternAcknowledged) {
+      toastNotifications.addSuccess(
+        i18n.translate(
+          'xpack.ml.dataframe.analyticsList.deleteAnalyticsWithIndexPatternSuccessMessage',
+          {
+            defaultMessage: 'Request to delete index pattern {destinationIndex} acknowledged.',
+            values: { destinationIndex },
+          }
+        )
+      );
+    }
+
+    if (Array.isArray(status.errors)) {
+      status.errors.map((error) => {
+        toastNotifications.addDanger(error.msg);
+      });
+    }
+  } catch (e) {
+    toastNotifications.addDanger(
+      i18n.translate('xpack.ml.dataframe.analyticsList.deleteAnalyticsErrorMessage', {
+        defaultMessage:
+          'An error occurred deleting the data frame analytics {analyticsId}: {error}',
+        values: { analyticsId: d.config.id, error: JSON.stringify(e) },
+      })
+    );
+  }
+  refreshAnalyticsList$.next(REFRESH_ANALYTICS_LIST_STATE.REFRESH);
+};
+
+export const checkUserCanDeleteIndex = async (indexName: string) => {
+  const toastNotifications = getToastNotifications();
+  try {
+    const privilege = await ml.hasPrivileges({
+      index: [
+        {
+          names: [indexName], // uses wildcard
+          privileges: ['delete_index'],
+        },
+      ],
+    });
+    return privilege.index[indexName].delete_index === true;
```

Comment:
this should also check for security being disabled and so could be:
```
return privilege.securityDisabled === true || privilege.has_all_requested === true;
```

Reply:
Done in https://github.com/elastic/kibana/pull/66934/commits/950f077b302daaaedee627f380733f78c3e376ea